docs/dev/cincinnati.md

```diff
@@ -0,0 +1,69 @@
+
+[google-omaha]: https://github.com/google/omaha/blob/v1.3.33.7/doc/ServerProtocolV3.md
+
+## Update Graph
+
+Cincinnati uses a [directed acyclic graph][dag] (DAG) to represent the valid updates.
+Each node in the graph is a release payload and each edge is a valid transition.
+
+[dag]: https://en.wikipedia.org/wiki/Directed_acyclic_graph
+
+## Client
+
+Cincinnati clients are the end consumers of the update graph and release payloads.
+A client periodically queries the Cincinnati server for updates and may apply them if available.
+
+## Graph API
+
+### Request
+
+HTTP `GET` requests are used to fetch the DAG (as a JSON object) from the Graph API endpoint.
+Requests SHOULD be sent to `/v1/graph` and MUST include the following header:
+
+```
+Accept: application/json
+```
+
+Fedora CoreOS clients MUST provide additional parameters as URL query parameters in the request.
+
+|        Key        | Optional | Description                                           |
+|-------------------|----------|-------------------------------------------------------|
+|      basearch     | required | base architecture (non-empty string)                  |
+|       stream      | required | client-selected update stream (non-empty string)      |
+|     node\_uuid    | optional | application-specific unique-identifier for the client |
+|    os\_version    | optional | current OS version                                    |
+|   os\_checksum    | optional | current OS checksum                                   |
+|       group       | optional | update group                                          |
+| rollout\_wariness | optional | client wariness to update rollout                     |
+|      platform     | optional | client platform                                       |
+
+```
+
+### Response
+
+A positive response to the `/v1/graph` endpoint MUST be a JSON representation of the release graph.
+Each of the releases are represented in an entry in the top-level `nodes` array.
+Each of these entries includes the release version label, a payload identifier and any metadata according to the following schema:
+
+|    Key   | Optional | Description                                                                             |
+|----------|----------|-----------------------------------------------------------------------------------------|
+| version  | required | the version of the release, as a unique (across "nodes" array) non-empty JSON string    |
+| payload  | required | payload identifier, as a JSON string                                                    |
+| metadata | required | a string-\>string map conveying arbitrary information to its consumers                  |
+
+The transitions between releases are represented as an array in the top-level `edges` array, where each entry is a tuple.
+Each of these tuples has two fields: the index of the starting node, and the index of the ending node. Both are non-negative integers, ranging from 0 to `len(nodes)-1`.
```

Comment:
Might be worth outputting a sample JSON graph just to show what it looks like.

Reply:
Ack, added as a separate file and referenced back here.